### ISSUE TITLE
Haskell layer -> hie does not exist as package

### ISSUE THREAD

**Issue by Martinsos** [syl20bnr/spacemacs#13249]
#### Description :octocat:
In instructions for installing haskell spacemacs layer, it says that we should install `hie` cabal package via cabal or stack.
However, there is no such package on either Stackage or Hackage, at least I was not able to find one!
I don't know if I am just not understanding something here, or documentation is outdated.

#### Reproduction guide :beetle:
- Go to https://github.com/syl20bnr/spacemacs/tree/develop/layers/+lang/haskell and read instructions for installation.
- Run `stack install hie` -> you will get error message.

*Observed behaviour:* :eyes: :broken_heart:
Got an error message.

*Expected behaviour:* :heart: :smile:
To install hie without any errors.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.3
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 96e7621f8)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(emacs-lisp helm multiple-cursors treemacs)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS GLIB NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 XDBE XIM MODULES THREADS LIBSYSTEMD LCMS2

#### Backtrace :paw_prints:
```
```

**Comment by ssbothwell** [comment]
Instructions for `HIE` installation on various operating systems is described here: https://github.com/haskell/haskell-ide-engine#installation

**Comment by Martinsos** [comment]
> Instructions for `HIE` installation on various operating systems is described here: https://github.com/haskell/haskell-ide-engine#installation

Sure, but then it would be better to point to those docs instead of saying that `hie` can be installed via cabal, right? I think that throws people off (it threw me off) -> I spend some time trying to figure out if there is package on hackage/stackage for hie that I should install, until I realized that there is no such package and that I need to install it manually.

**Comment by ssbothwell** [comment]
The only mention of `HIE` is in the `LSP` section here:

**Comment by Martinsos** [comment]
I was referring to this part of the README.org in https://github.com/syl20bnr/spacemacs/tree/develop/layers/+lang/haskell:

```
This layer requires some cabal packages:

apply-refact (required by hlint-refactor)
hlint (required by hlint-refactor)
stylish-haskell (optional for haskell-mode)
hasktags (optional)
hoogle (optional for haskell-mode and helm-hoogle)
ghc-mod (optional for completion)
intero (optional for completion)
hie (optional for completion)

To install them, use the following command (or the stack equivalent):

$ cabal install apply-refact hlint stylish-haskell hasktags hoogle
```

From this, it reads that hie is cabal package -> that is what caused confusion in my case.

**Comment by ssbothwell** [comment]
Ahh, I can see how that would be a bit confusing. `HIE` is a large project that has to be installed in particular ways depending on your operating system. 

Generally there are a bunch of ways to install haskell packages that are often operating system dependent. It would probably be a good idea to replace this section with links to the instructions for each of these packages. I'm happy to do that some time this week.

**Comment by Martinsos** [comment]
No problem, I think just removing hie from this list and mentioning it
below as separate, system dependency, would be enough, even without link to
installation instructions. Thanks!

**Comment by smile13241324** [comment]
I have had a look at the hie installation instructions and they were really a bit lacking. I have added a link to the `hie` installation instruction now and also removed `hie` from the initial list of cabal packages to install on develop.

@Martinsos could you double check whether this would make matters more clear, and close the ticket if it does?

Btw: What's your opinion about the amount of backend choices in the haskell layer, does it come clear which backend to choose in which situation or is the amount of supported backends more confusing?

**Comment by Martinsos** [comment]
Great, I think that is it, it should be clear now (at least to me)! I will
close the issue.

I already have experience from before, so for me it was relatively clear
which backend does what.
I was using Intero before, in vanilla emacs, but started looking for smth
else since Intero is not maintained any more (keeping that in mind, it will
probably make sense to remove it from the list of backends at some point?).
I tried using Dante in my vanilla emacs config, but it did not work out of
the box and I did not set it up at the end.
Same with lsp -> I mostly got it working in vanilla emacs, but it was still
not as good as Intero.
That is why I switched to spacemacs, and was looking forward to haskell
layer ;)!
However, I have to say I still have same problems with haskell lsp in
spacemacs as I had in my vanilla emacs -> spurious errors, type and
auto-completion information showing in too small popup windows so it is not
usable. That is why I am currently using `ghci` backend, but will soon try
experimenting again -> I will probably try to use Dante or Lsp again, I
just have to put more time into researching how to configure them.
I think it is good taht you made it clear what the default option is, for
backend, that should be enough for somebody new.
